app.ts

```diff
@@ -0,0 +1,131 @@
+enum Choice {
+    Rock = 'Rock',
+    Paper = 'Paper',
+    Scissors = 'Scissors'
+}
+
+enum Result {
+    Win,
+    Loss,
+    Draw
+}
+
+
+const ChoiceResultMap = {
+    [Choice.Rock]: {
+        [Choice.Scissors]: Result.Win,
+        [Choice.Paper]: Result.Loss,
+    },
+    [Choice.Paper]: {
+        [Choice.Rock]: Result.Win,
+        [Choice.Scissors]: Result.Loss,
+    },
+    [Choice.Scissors]: {
+        [Choice.Paper]: Result.Win,
+        [Choice.Rock]: Result.Loss,
+    }
+}
+
+/**
+ * Returns the members that an enum has
+ * @param e An Enum
+ * @returns Count of individual choices in enum
+ */
+const getMembersOfEnum = (e): number => {
+    return Object.keys(e).length;
+}
```

Comment:
Some languages can tell you how many members an enum has, but javascript doesn't. Boo. So this is a nice utility